Lombiq.HelpfulLibraries.LinqToDb/LinqToDbQueryExecutor.cs

```diff
@@ -0,0 +1,59 @@
+using Dapper;
+using LinqToDB;
+using LinqToDB.Data;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace Lombiq.HelpfulLibraries.LinqToDb
+{
+    public static class LinqToDbQueryExecutor
+    {
+        public static async Task<IEnumerable<TResult>> LinqQueryAsync<TResult>(
+            this ISession session,
+            Func<ITableAccessor, IQueryable> query)
+        {
+            var transaction = await session.DemandAsync();
+            var convertedSql = ConvertSqlToDialect(session, transaction, query);
+
+            return await transaction.Connection.QueryAsync<TResult>(convertedSql, transaction: transaction);
+        }
+
+        private static string ConvertSqlToDialect(
+            ISession session,
+            IDbTransaction transaction,
+            Func<ITableAccessor, IQueryable> query)
+        {
+            // Generate aliases for final projection.
+            LinqToDB.Common.Configuration.Sql.GenerateFinalAliases = true;
+
+            // Instantiating a linq2db connection object as it is required to start building the query. Note that it
+            // won't create an actual connection with the database yet.
+            var dataProvider = DataConnection.GetDataProvider(
+                GetDatabaseProviderName(session.Store.Dialect.Name),
+                transaction.Connection.ConnectionString);
+
+            using var linqToDbConnection = new LinqToDbConnection(
+                dataProvider,
+                transaction,
+                session.Store.Configuration.TablePrefix);
+            return query(linqToDbConnection).ToString();

```

Comment:
Csak, hogy biztosra menjünk, mert nagyon fontos, és most jelentősen változott a kód: ez továbbra is paramétereket hoz létre az inputoknak SQL-ben, ugye?

Reply:
Igen. Amikor kérdezted akkor volt ebben a formában igazából.